src/components/pages/Triggers/TriggersList/AddTriggersModal/AddTriggerModal.tsx

```diff
@@ -0,0 +1,179 @@
+
+import {useCreateTriggerMutation} from '@services/triggers';
+
+import {displayDefaultNotificationFlow} from '@src/utils/notification';
+
+import Colors from '@styles/Colors';
+
+import {getResourceIdentifierSelector} from '../../utils';
+
+import {StyledButtonsContainer, StyledNotificationContainer, StyledStepDescription} from './AddTriggersModal.styled';
+
+type AddSourceFormValues = {
+  name: string;
+  uri: string;
+  token?: string;
+  username?: string;
+};
+
+const AddTriggerModal: React.FC = () => {
+  const {navigate} = useContext(DashboardContext);
+
+  const [createTrigger, {isLoading}] = useCreateTriggerMutation();
+
+  const appNamespace = useAppSelector(selectNamespace);
+
+  const [error, setError] = useState<ErrorNotificationConfig | undefined>(undefined);
+  const [currentStep, setCurrentStep] = useState(0);
+  const [name, setName] = useState('');
+  const [firstStepValues, setFirstStepValues] = useState<Record<string, string | Option[]>>({});
+
+  const topRef = useRef<HTMLDivElement>(null);
+  const inTopInViewport = useInViewport(topRef);
+
+  const [form] = Form.useForm();
+
+  const onFinish = () => {
+    const values = form.getFieldsValue();
+
+    const resourceSelector = getResourceIdentifierSelector(
+      firstStepValues.resourceLabelSelector || firstStepValues.resourceNameSelector,
+      appNamespace
+    );
+
+    const testSelector = getResourceIdentifierSelector(
+      values.testLabelSelector || values.testNameSelector,
+      appNamespace
+    );
+
+    const [action, execution] = values.action.split(' ');
+
+    const body = {
+      ...(name ? {name} : {}),
+      resource: firstStepValues.resource,
+      event: firstStepValues.event,
+      action,
+      execution,
+      testSelector,
+      resourceSelector,
+    };
+    createTrigger(body)
+      .then(res => displayDefaultNotificationFlow(res))
+      .then(res => {
+        if (res && 'data' in res) {
+          navigate(`/triggers/${res.data.name}`);
+        }
+      })
+      .catch(err => {
+        setError(err);
+
+        if (!inTopInViewport && topRef && topRef.current) {
+          topRef.current.scrollIntoView();
+        }
+      });
+  };
+
+  return (
+    <>
+      <div ref={topRef} />
+      <Space size={24} direction="vertical" style={{width: '100%', marginTop: 0}}>
+        <Space direction="vertical" style={{width: '100%'}}>
+          {error ? (
+            <StyledNotificationContainer>
+              <NotificationContent status="failed" message={error.message} title={error.title} />
+            </StyledNotificationContainer>
+          ) : null}
+          <Text className="regular middle">Name</Text>
+          <Input
+            placeholder="e.g.: container-deployment-xyz"
+            value={name}
+            onChange={event => {
+              setName(event.target.value);
+            }}
+          />
+        </Space>
+        <Steps current={currentStep} items={[{title: 'Condition'}, {title: 'Action'}]} />
```

Comment:
Is Antd step what we want, or should we move steps from Cloud implementation?

Reply:
Yes, it is exactly what we want